src/platform_impl/linux/global_shortcut.rs

```diff
@@ -1,389 +1,110 @@
+      ))
     }
-
-    Err(ShortcutManagerError::InvalidAccelerator(
-      "Invalid accelerators".into(),
-    ))
   }
 
   pub(crate) fn unregister_all(&mut self) -> Result<(), ShortcutManagerError> {
-    for (found_id, _) in self.shortcuts.lock().unwrap().iter() {
-      self
-        .method_sender
-        .send(HotkeyMessage::UnregisterHotkey(*found_id))
-        .map_err(|_| ShortcutManagerError::InvalidAccelerator("Channel error".into()))?;
+    if let Err(e) = self.tx.send(ShortcutEvent::UnRegisterAll) {
+      log::warn!(
+        "Failed to send global shortcut event to event loop channel: {}",
+        e
+      );
     }
-    self.shortcuts = ListenerMap::default();
     Ok(())
   }
 
   pub(crate) fn unregister(
     &self,
     shortcut: RootGlobalShortcut,
   ) -> Result<(), ShortcutManagerError> {
-    let mut found_id = (-1, 0);
-    for (id, shortcut_id) in self.shortcuts.lock().unwrap().iter() {
-      if *shortcut_id == shortcut.0.id().0 as u32 {
-        found_id = *id;
-        break;
-      }
-    }
-    if found_id == (-1, 0) {
-      return Err(ShortcutManagerError::AcceleratorNotRegistered(
-        shortcut.0.accelerator,
-      ));
-    }
-
-    self
-      .method_sender
-      .send(HotkeyMessage::UnregisterHotkey(found_id))
-      .map_err(|_| ShortcutManagerError::InvalidAccelerator("Channel error".into()))?;
-    if self.shortcuts.lock().unwrap().remove(&found_id).is_none() {
-      panic!("hotkey should never be none")
-    };
-    match self.method_receiver.recv() {
-      Ok(HotkeyMessage::UnregisterHotkeyResult(Ok(_))) => Ok(()),
-      Ok(HotkeyMessage::UnregisterHotkeyResult(Err(err))) => Err(err),
-      Err(err) => Err(ShortcutManagerError::InvalidAccelerator(err.to_string())),
-      _ => Err(ShortcutManagerError::InvalidAccelerator(
-        "Unknown error".into(),
-      )),
-    }
-  }
-}
-
-impl Drop for ShortcutManager {
-  fn drop(&mut self) {
-    if let Err(err) = self.method_sender.send(HotkeyMessage::DropThread) {
-      #[cfg(debug_assertions)]
-      eprintln!("cant send close thread message {}", err);
+    if let Err(e) = self
+      .tx
+      .send(ShortcutEvent::UnRegister(shortcut.0.accelerator.id()))
+    {
+      log::warn!(
+        "Failed to send global shortcut event to event loop channel: {}",
+        e
+      );
     }
+    Ok(())
   }
 }
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct GlobalShortcut {
   pub(crate) accelerator: Accelerator,
 }
-type ListenerId = (i32, u32);
-type ListenerMap = Arc<Mutex<HashMap<ListenerId, u32>>>;
 
 impl GlobalShortcut {
   pub fn id(&self) -> AcceleratorId {
     self.accelerator.clone().id()
   }
 }
 
-// required for event but we use dummy window Id
-// so it shouldn't be a problem
-unsafe impl Send for WindowId {}
-unsafe impl Sync for WindowId {}
-// simple enum, no pointer, shouldn't be a problem
-// to use send + sync
-unsafe impl Send for WindowRequest {}
-unsafe impl Sync for WindowRequest {}
+pub enum ShortcutEvent {
```

Comment:
```suggestion
pub enum GloablShortcutRequest {
```
to align with `WindowRequest` enum. Also the error messages should reflect this change.